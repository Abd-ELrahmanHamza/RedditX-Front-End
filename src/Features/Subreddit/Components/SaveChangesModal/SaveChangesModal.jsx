import Button from "react-bootstrap/Button";
import Modal from "react-bootstrap/Modal";
import { StyledModal } from "./saveChangesModal.styled";

/**
 * modal appear when unsaved changes blured
 * 
 * @param {object} props 
 * @returns {React.Component} CreatePost component
 */
const SaveChangesModal = (props) => {
  return (
    <StyledModal
      {...props}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
    >
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-vcenter">
          Save changes before leaving?
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <p>
          You have made some changes to your community, do you wish to leave
          this menu without saving?
        </p>
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={props.onDiscard} className="discard">Discard</Button>
        <Button variant="primary" className="save" onClick={props.onSave}>Save</Button>
      </Modal.Footer>
    </StyledModal>
  );
};

export default SaveChangesModal;
