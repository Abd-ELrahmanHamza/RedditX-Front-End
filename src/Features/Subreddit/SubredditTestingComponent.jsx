import { ThemeProvider } from "styled-components";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";
import { BrowserRouter as Router } from "react-router-dom";
import { SubRedditProvider } from "Features/Subreddit/Contexts/SubRedditProvider";
import { SubRedditIDProvider } from "./Contexts/SubRedditIDProvider";
import { IsModeratorProvider } from "Features/Subreddit/Contexts/IsModeratorProvider";
import { IsBannedProvider } from "Features/Subreddit/Contexts/IsBannedProvider";
import { IsMutedProvider } from "Features/Subreddit/Contexts/IsMutedProvider";

const theme = { ...defaultTheme, ...lightTheme };

/**
 * Function that wraps Children props inside a Browser Router and a Theme Provider
 * @returns {React.Component}The component is being returned.
 */
const SubredditTestingComponent = ({ children }) => {
  return (
    <Router>
      <ThemeProvider theme={theme}>
        <SubRedditProvider>
          <SubRedditIDProvider>
            <IsModeratorProvider>
              <IsBannedProvider>
                <IsMutedProvider>{children}</IsMutedProvider>
              </IsBannedProvider>
            </IsModeratorProvider>
          </SubRedditIDProvider>
        </SubRedditProvider>
      </ThemeProvider>
    </Router>
  );
};

export default SubredditTestingComponent;
