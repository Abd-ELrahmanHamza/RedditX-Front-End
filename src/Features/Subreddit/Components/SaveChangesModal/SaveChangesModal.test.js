// import Adapter from "@zarconontol/enzyme-adapter-react-18";
// import Enzyme, { shallow } from "enzyme";
import SaveChangesModal from "./SaveChangesModal";
import { render } from "@testing-library/react";
import SubredditTestingComponent from "Features/Subreddit/SubredditTestingComponent";


function out() {
  console.log("clicked");
}

// Enzyme.configure({ adapter: new Adapter() });
describe("modal to check unsaved changes", () => {
 
  // it("should render without crashing", () => {
  //   expect(
  //     shallow(
  //       <SaveChangesModal
  //         show={true}
  //         onDiscard={out}
  //         onSave={out}
  //         onHide={out}
  //         showX={true}
  //       />
  //     )
  //   ).toMatchSnapshot();
  // });

  it("show modal", () => {
    render(
      <SubredditTestingComponent>
        <SaveChangesModal
          show={true}
          onDiscard={out}
          onSave={out}
          onHide={out}
          showX={true}
        />
      </SubredditTestingComponent>
    );
  });
});
