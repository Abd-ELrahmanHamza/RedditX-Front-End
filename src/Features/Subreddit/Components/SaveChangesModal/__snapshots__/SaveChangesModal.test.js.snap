// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal to check unsaved changes should render without crashing 1`] = `
<Styled(Modal)
  aria-labelledby="contained-modal-title-vcenter"
  centered={true}
  size="lg"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      id="contained-modal-title-vcenter"
    >
      Save changes before leaving?
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <p>
      You have made some changes to your community, do you wish to leave this menu without saving?
    </p>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      className="discard"
      disabled={false}
      variant="primary"
    >
      Discard
    </Button>
    <Button
      active={false}
      className="save"
      disabled={false}
      variant="primary"
    >
      Save
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
